rule.php

```diff
@@ -62,6 +65,39 @@ public static function make (quiz $quizobj, $timenow, $canignoretimelimits) {
      * @param MoodleQuickForm $mform the wrapped MoodleQuickForm.
      */
     public static function add_settings_form_fields(mod_quiz_mod_form $quizform, MoodleQuickForm $mform) {
+        global $DB, $USER;
+        $settingsmanager = new settings_manager();
```

Comment:
Love how you abstract all fields in the manager. :100:  Made usually messy form code more readable